components/humantask/org.wso2.carbon.humantask/src/main/java/org/wso2/carbon/humantask/core/scheduler/NotificationScheduler.java

```diff
@@ -0,0 +1,272 @@
+
+import org.wso2.carbon.event.output.adapter.core.OutputEventAdapterConfiguration;
+import org.wso2.carbon.event.output.adapter.core.exception.OutputEventAdapterException;
+import org.wso2.carbon.event.output.adapter.email.EmailEventAdapter;
+import org.wso2.carbon.event.output.adapter.sms.SMSEventAdapter;
+import org.wso2.carbon.humantask.TRendering;
+import org.wso2.carbon.humantask.core.HumanTaskConstants;
+import org.wso2.carbon.humantask.core.dao.TaskCreationContext;
+import org.wso2.carbon.humantask.core.dao.TaskDAO;
+import org.wso2.carbon.humantask.core.engine.runtime.api.EvaluationContext;
+import org.wso2.carbon.humantask.core.engine.runtime.api.ExpressionLanguageRuntime;
+import org.wso2.carbon.humantask.core.engine.util.CommonTaskUtil;
+import org.wso2.carbon.humantask.core.internal.HumanTaskServiceComponent;
+import org.wso2.carbon.humantask.core.store.HumanTaskBaseConfiguration;
+import org.xml.sax.InputSource;
+import org.w3c.dom.Document;
+
+import javax.xml.namespace.QName;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import java.io.StringReader;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ExecutorService;
+
+
+public class NotificationScheduler {
+    private static Log log = LogFactory.getLog(NotificationScheduler.class);
+    private boolean isNotification = false;
+    private boolean isEmailNotification = false;
+    private boolean isSMSNotification = false;
+    private boolean taskCompleted = false;
+    private static EmailEventAdapter emailAdapter;
+    private static SMSEventAdapter smsAdapter;
+    private NotificationScheduler notificationScheduler;
+    private ExecutorService exec;
+    private NotificationTask notificationTask;
+    private Map<String, String> emailDynamicProperties;
+    private Map<String, String> smsDynamicProperties;
+
+
+    private String prefix = "htd";
+    String mail_to = null;
+    //String ccEmail = null;
+    // String bccEmail = null;
+    String mail_subject = null;
+    String emailBody = null;
+    String emailBodyTagVal = null;
+    String smsReceiver = null;
+    String smsBodyTagVal = null;
+    String smsBody = null;
+
+
+    public void setExecutorService(ExecutorService executorService) {
+        exec = executorService;
+    }
+
+
+    public static synchronized EmailEventAdapter getEmailEventAdapter(Map<String, String> dynamicProperties) {
+        if (emailAdapter == null) {
+            emailAdapter = new EmailEventAdapter(null, dynamicProperties);
+        }
+        try {
+            emailAdapter.init();
+        } catch (OutputEventAdapterException e) {
+             log.error("unable to initialize adapter"+ emailAdapter + e);
+
+        }
+
+        return emailAdapter;
+    }
+
+    public static synchronized SMSEventAdapter getSMSEventAdapter(Map<String, String> dynamicProperties) {
+        if (smsAdapter == null) {
+            //Need staticProperties at init()
+
```

Comment:
Defined string constants for properties of SMSEventAdapter in commit https://github.com/wso2/carbon-business-process/commit/c037c2607734679028dc96c59761857176b6dfdb